server/src/test/java/io/unitycatalog/server/sdk/access/SdkAccessControlBaseCRUDTest.java

```diff
@@ -0,0 +1,286 @@
+    // Revoke granted permissions
+    for (PermissionGrant grant : grantedPermissions) {
+      try {
+        LOGGER.debug(
+            "Revoking permissions {} from {} on {}",
+            grant.privileges,
+            grant.userEmail,
+            grant.resourceFullName);
+        PermissionsChange permissionsChange =
+            new PermissionsChange()
+                .principal(grant.userEmail)
+                .add(List.of())
+                .remove(grant.privileges);
+        UpdatePermissions updatePermissions =
+            new UpdatePermissions().changes(java.util.List.of(permissionsChange));
+        grantsApi.update(grant.resourceSecurableType, grant.resourceFullName, updatePermissions);
+      } catch (Exception e) {
+        LOGGER.warn(
+            "Failed to revoke permissions for {} on {}: {}",
+            grant.userEmail,
+            grant.resourceFullName,
+            e.getMessage());
+      }
+    }
+
+    // Delete created users
+    for (String userId : createdUserIds) {
+      try {
+        LOGGER.debug("Deleting test user: {}", userId);
+        usersApi.deleteUser(userId);
+      } catch (Exception e) {
+        LOGGER.warn("Failed to delete test user {}: {}", userId, e.getMessage());
+      }
+    }
+
+    // Clear tracking lists for each test
+    createdUserIds.clear();
+    grantedPermissions.clear();
+  }
+
+  /** Creates common resources (catalog and schema) using an admin token for authorization. */
+  private void createCommonResources() {
+    CatalogOperations adminCatalogOps =
+        new SdkCatalogOperations(TestUtils.createApiClient(adminConfig));
+    SchemaOperations adminSchemaOps =
+        new SdkSchemaOperations(TestUtils.createApiClient(adminConfig));
+
+    CreateCatalog createCatalog =
+        new CreateCatalog().name(TestUtils.CATALOG_NAME).comment(TestUtils.COMMENT);
+    try {
+      adminCatalogOps.createCatalog(createCatalog);
+      adminSchemaOps.createSchema(
+          new CreateSchema().name(TestUtils.SCHEMA_NAME).catalogName(TestUtils.CATALOG_NAME));
+    } catch (ApiException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Creates a control API client for user management operations.
+   *
+   * @param config Server configuration including URL and auth token
+   * @return A configured control API client
+   */
+  protected io.unitycatalog.control.ApiClient createControlApiClient(ServerConfig config) {
+    io.unitycatalog.control.ApiClient controlApiClient = new io.unitycatalog.control.ApiClient();
+    java.net.URI uri = java.net.URI.create(config.getServerUrl());
+    int port = uri.getPort();
+    controlApiClient.setHost(uri.getHost());
+    controlApiClient.setPort(port);
+    controlApiClient.setScheme(uri.getScheme());
+    if (config.getAuthToken() != null && !config.getAuthToken().isEmpty()) {
+      controlApiClient.setRequestInterceptor(
+          request -> request.header("Authorization", "Bearer " + config.getAuthToken()));
+    }
```

Comment:
I've seen lots of places are trying to constrcut the api client by themselves.  Actually we need to unify them into one place, because: 

1.  Retry policy. 
2.  need to care about this issue: https://github.com/unitycatalog/unitycatalog/pull/1130